### PR TITLE
Add applyAddon typings

### PR DESCRIPTION
**Pull request by Tyriar**
Part of #1018
See #1130

### CODE REVIEW

File: typings/xterm.d.ts [hunk 1]
```diff
@@ -504,6 +504,6 @@ declare module 'xterm' {
      * available to all newly created Terminals.
```
Comment (simark) [inline typings/xterm.d.ts]: Should the wording in the comment be updated too?

**Comment by simark** [comment]
LGTM.

**Comment by parisk** [comment]
@Tyriar I think we should update the `typings-test.ts` also: https://github.com/xtermjs/xterm.js/blob/v3/fixtures/typings-test/typings-test.ts#L37-L46

## Build error
```
stdout: typings-test.ts(39,14): error TS2339: Property 'loadAddon' does not exist on type 'typeof Terminal'.
typings-test.ts(40,14): error TS2339: Property 'loadAddon' does not exist on type 'typeof Terminal'.
typings-test.ts(41,14): error TS2339: Property 'loadAddon' does not exist on type 'typeof Terminal'.
typings-test.ts(42,14): error TS2339: Property 'loadAddon' does not exist on type 'typeof Terminal'.
typings-test.ts(43,14): error TS2339: Property 'loadAddon' does not exist on type 'typeof Terminal'.
typings-test.ts(44,14): error TS2339: Property 'loadAddon' does not exist on type 'typeof Terminal'.
```

**Comment by Tyriar** [comment]
Glad to see the tests are doing their job 😄

**Pull request merged**